lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1439,9 +1439,18 @@ static LogicalResult foldSingleSetConnect(ConnectOp op,
   if (srcValueOp && srcValueOp != &declBlock->front())
     srcValueOp->moveBefore(&declBlock->front());
```

Comment:
We don't need to move the constant if we're replacing not going to use it

Reply:
Great catch. I'll move this to an `else if` condition following the check on `InvalidValueOp` below.

Reply:
Fixed in the force push.